### PR TITLE
Validation with regex

### PR DESCRIPTION
**Pull request by PijusOz**
Separators in date can be space, nothing, '-' or '/'.  Illogical month or day number is considered.
'.' in amount is optional, however after it only 1 or 2 numbers are allowed.

### CODE REVIEW

File: front-end-side/src/Pages/EnterData.js [hunk 1]
```diff
@@ -8,23 +8,21 @@ const ShowData = () => {
...
-    const handleSubmit = async(e) => {
-        e.preventDefault();
-        await axios.get('https://localhost:7174/api/ShowData').then(resp => {
-            setInfo(resp);
-            setStatus(true);
-        })
-    }
-    const show = () => {
-        setStatus(true);
+        const dateRegEx = /^([0-9]{4}[-/ ]?((0[13-9]|1[012])[-/ ]?(0[1-9]|[12][0-9]|30)|(0[13578]|1[02])[-/ ]?31|02[-/ ]?(0[1-9]|1[0-9]|2[0-8]))|([0-9]{2}(([2468][048]|[02468][48])|[13579][26])|([13579][26]|[02468][048]|0[0-9]|1[0-6])00)[-/ ]?02[-/ ]?29)$/
```
Comment (dissco6) [inline front-end-side/src/Pages/EnterData.js:14]: Can we make that date 2022-5-6 also would be valid?

File: front-end-side/src/Pages/EnterData.js [hunk 2]
```diff
@@ -8,23 +8,21 @@ const ShowData = () => {
...
+        const amountRegEx = /^\d+\.?\d{1,2}$/
+        if (!dateRegEx.test(date)) {
+            alert("Date is invalid");
+        }
```
Comment (dissco6) [inline front-end-side/src/Pages/EnterData.js:18]: Also don't let to leave Details and Seller empty

File: front-end-side/src/Pages/EnterData.js [hunk 3]
```diff
@@ -8,23 +8,21 @@ const ShowData = () => {
...
+        else if (!amountRegEx.test(amount)) {
+            alert("Amount is invalid");
```
Comment (dissco6) [inline front-end-side/src/Pages/EnterData.js:20]: Can we show error not as an alert, but as text in our page?

File: front-end-side/src/Pages/EnterData.js (hunk 1 above)
Comment (PijusOz) [reply front-end-side/src/Pages/EnterData.js:14]: Done ✅

File: front-end-side/src/Pages/EnterData.js (hunk 2 above)
Comment (PijusOz) [reply front-end-side/src/Pages/EnterData.js]: Done ✅

File: front-end-side/src/Pages/EnterData.js (hunk 3 above)
Comment (PijusOz) [reply front-end-side/src/Pages/EnterData.js]: Done ✅

File: front-end-side/src/Pages/EnterData.css [hunk 4]
```diff
@@ -8,4 +8,11 @@
 #form button {
     grid-column-start: 4;
     grid-row-start: 2;
+}
+
+p#error-msg {
```
Comment (dissco6) [inline front-end-side/src/Pages/EnterData.css:13]: Add more precise path (classes and ids) to ``p#error-msg``

**Pull request merged**